include/boost/wave/util/cpp_macromap_utils.hpp

```diff
@@ -142,7 +142,7 @@ is_special_macroname (StringT const &name)
     if (name.size() < 7)
         return false;
 
-    if ("defined" == name)
+    if (("defined" == name) || ("__has_include" == name))
```

Comment:
Should we check for `__has_include` only if it's detection was enabled (C++17)?

Reply:
Yes :) I am actually writing the feature tests (and conditional compilation stuff) right now.
Also I think this is a C++17 feature - so I'm adding a C++17 mode as well.

Reply:
Ok, C++17, then. In this case you might need to adapt the tests to ask for this as well.